sources/bloodhound/src/lib.rs

```diff
@@ -13,7 +13,57 @@ pub fn look_for_string_in_file(path: &str, search_str: &str) -> Result<bool, io:
         .any(|line| line.unwrap_or_default().contains(search_str)))
 }
 
-/// Executes a command and checks if the give `search_str` is in the output.
+/// Reads a file and checks if the given `search_str` is present in its contents.
+pub fn look_for_strings_in_file(path: &str, search_strs: &[&str]) -> Result<bool, io::Error> {
+    let mut matched = 0;
+
+    let reader = BufReader::new(File::open(path)?);
+    for line in reader.lines() {
+        let content = line.unwrap_or_default();
+        for search_str in search_strs {
+            if content.contains(search_str) {
+                matched += 1;
+            }
```

Comment:
You could return earlier if an expected string isn't found.

Reply:
I'm not seeing a good way to exit early here. We need to go through all lines to see if the strings are present. And we need to check each string on each line because the one line could contain multiple substrings that we are trying to match.

If you have something in mind that I'm missing, can you add some pseudo code to describe it?